sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -7589,7 +7595,12 @@ public void testProjectAfterSort2() throws Exception
     );
   }
 
+  /**
+   * In Calcite 1.17, this test worked, but after upgrading to Calcite 1.21, this query fails with:
```

Comment:
Please do two things in response to this:

1. Keep this test as-is, but add a new, second test that is the same basic query shape but doesn't trip the `dim1` is ambiguous thing. Maybe rename one of them, or group by ordinals. Testing project after sort is still important, so there's value in this.
2. If you think this is a Calcite bug, raise a Calcite issue at: https://issues.apache.org/jira/projects/CALCITE/

Reply:
I added a new test:

```
 @Test
  public void testProjectAfterSort3WithoutAmbiguity() throws Exception
  {
    // This query is equivalent to the one in testProjectAfterSort3 but renames the second grouping column
    // to avoid the ambiguous name exception. The inner sort is also optimized out in Calcite 1.21.
    testQuery(
        "select copydim1 from (select dim1, dim1 AS copydim1, count(*) cnt from druid.foo group by dim1, dim1 order by cnt)",
        ImmutableList.of(
            GroupByQuery.builder()
                        .setDataSource(CalciteTests.DATASOURCE1)
                        .setInterval(querySegmentSpec(Filtration.eternity()))
                        .setGranularity(Granularities.ALL)
                        .setDimensions(
                            dimensions(
                                new DefaultDimensionSpec("dim1", "d0")
                            )
                        )
                        .setContext(QUERY_CONTEXT_DEFAULT)
                        .build()
        ),
        ImmutableList.of(
            new Object[]{""},
            new Object[]{"1"},
            new Object[]{"10.1"},
            new Object[]{"2"},
            new Object[]{"abc"},
            new Object[]{"def"}
        )
    );
  }
```

As with the other project-after-sort tests, the inner ordering is optimized out.

I'll need to do some more investigation to see if this is a bug or not.